File: app/src/main/java/io/zenandroid/greenfield/dagger/Injector.java
package io.zenandroid.greenfield.dagger;

/**
 * Created by acristescu on 22/06/2017.
 */

import android.content.Context;
import android.support.annotation.VisibleForTesting;

public class Injector {

	private static AppComponent component;

	public static void init(Context context) {
		if (component == null) {
			component = DaggerAppComponent.builder().appModule(new AppModule()).build();
		}
	}

	public static AppComponent get() {
		return component;
	}

	@VisibleForTesting
	public static void setComponent(AppComponent component) {
		Injector.component = component;
	}
}

File: app/src/main/java/io/zenandroid/greenfield/dagger/Injector.kt
package io.zenandroid.greenfield.dagger

/**
 * Created by acristescu on 22/06/2017.
 */

import android.content.Context
import androidx.annotation.VisibleForTesting

object Injector {

    private lateinit var component: AppComponent

    fun init(context: Context) {
        component = DaggerAppComponent.builder().appModule(AppModule()).build()
    }

    fun get(): AppComponent {
        return component
    }

    @VisibleForTesting
    fun setComponent(component: AppComponent) {
        Injector.component = component
    }
}
